modules/phase_field/src/materials/InterfaceOrientationMaterial.C

```diff
@@ -32,15 +42,37 @@ void
 InterfaceOrientationMaterial::computeQpProperties()
 {
   // cosine of the gradient orientation angle
-  Real cos_grad;
-  const Real grad2 = _grad_u[_qp] * _grad_u[_qp];
-  if (grad2 == 0)
-    cos_grad = 0;
+  Real n;
+  if (_grad_op[_qp].norm() == 0)
+    n = 0;
   else
-    cos_grad = _grad_u[_qp](0) / std::sqrt(grad2);
+    n = _grad_op[_qp](0) / _grad_op[_qp].norm();
 
-  const Real angle = std::acos(cos_grad);
+  if (n > 0.99999)
+    n = 0.99999;
 
-  _eps[_qp]= 0.01 * (1.0 + 0.04 * std::cos(6 * (angle - libMesh::pi/2.0)));
-  _deps[_qp]= -0.01 * 0.04 * 6.0 * std::sin(6 * (angle - libMesh::pi/2.0));
+  const Real angle = std::acos(n);
+
+  // Calculate interfacial parameter epsilon and its derivatives
+  _eps[_qp]= _eps_bar * (_delta * std::cos(_j * (angle - _theta0 * libMesh::pi/180.0)) + 1.0);
+  _deps[_qp]= - _eps_bar * _delta * _delta * std::sin(_j * (angle - _theta0 * libMesh::pi/180.0));
+  Real d2eps = - _eps_bar * _delta * _delta * _delta * std::cos(_j * (angle - _theta0 * libMesh::pi/180.0));
+
+  // Compute derivative of n with respect to grad_op
+  RealGradient dndgrad_op;
+  if (_grad_op[_qp].norm_sq() == 0)
+    dndgrad_op = 0;
+  else
+  {
+    dndgrad_op(0) = _grad_op[_qp](1) * _grad_op[_qp](1);
+    dndgrad_op(1) = - _grad_op[_qp](0) * _grad_op[_qp](1);
```

Comment:
Just making sure that the second factor is supposed to have an index 1 in both lines...